soc/spid.c

```diff
@@ -97,6 +102,114 @@ void delay() {
     while(i--);
 }
 
+/*
+Fetches the stored open/closed state of a given channel
+
+Args:
+    - channel: the index of the channel to check the status of
+    - channels_open: the bit flag containing channel status
+
+Returns:
+    STATUS_OPENED if state is currently open
+    STATUS_CLOSED if state is currently closed
+
+*/
+uint8_t get_channel_status(uint8_t channel, uint8_t *channels_open) {
```

Comment:
Might as well make the channel mask variable(s) global if you're going to break out these functions.